include/flatbuffers/flatbuffers.h

```diff
@@ -445,32 +445,26 @@ class DetachedBuffer {
     : allocator_(other.allocator_), own_allocator_(other.own_allocator_),
       buf_(other.buf_), reserved_(other.reserved_), cur_(other.cur_),
       size_(other.size_) {
-    other.allocator_ = nullptr;
-    other.own_allocator_ = false;
-    other.buf_ = nullptr;
-    other.reserved_ = 0;
-    other.cur_ = nullptr;
-    other.size_ = 0;
+    other.reset();  
   }
 
   DetachedBuffer &operator=(DetachedBuffer &&other) {
-    std::swap(allocator_, other.allocator_);
-    std::swap(own_allocator_, other.own_allocator_);
-    std::swap(buf_, other.buf_);
-    std::swap(reserved_, other.reserved_);
-    std::swap(cur_, other.cur_);
-    std::swap(size_, other.size_);
+    destroy();
+
+    allocator_ = other.allocator_;
+    own_allocator_ = other.own_allocator_;
+    buf_ = other.buf_;
+    reserved_ = other.reserved_;
+    cur_ = other.cur_;
+    size_ = other.size_;
+
+    other.reset();
+
```

Comment:
We could minimize the diff by keeping the swaps and just calling `other.destroy()` afterwards. For me that reads a little more directly. What do you think?

Reply:
Personally not fond of `std::swap` in move assignments.